src/python/pants/backend/cc/subsystems/compiler.py

```diff
@@ -0,0 +1,140 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+import logging
+from typing import Iterable
+
+from pants.core.util_rules.external_tool import TemplatedExternalTool
+from pants.engine.rules import Rule, collect_rules
+from pants.engine.unions import UnionRule
+from pants.option.option_types import StrListOption, StrOption
+from pants.option.subsystem import Subsystem
+from pants.util.strutil import softwrap
+
+logger = logging.getLogger(__name__)
+
+
+class CCOptions:
+    c_compile_options = StrListOption(
+        default=["-std=c11"],
+        help=softwrap(
+            """
+            Flags passed to the C compiler.
+            These flags are merged with the target-level defines, with the target-level flags taking precedence.
+            """
+        ),
+    )
+
+    c_defines = StrListOption(
+        default=None,
+        help=softwrap(
+            """
+            A list of strings to define in the C preprocessor. Will be prefixed by -D at the command line.
+            These defines are merged with the target-level defines, with the target-level definitions taking precedence.
+            """
+        ),
+    )
+
+    cpp_compile_options = StrListOption(
+        default=["-std=c++11"],
+        help=softwrap(
+            """
+            Flags passed to the C++ compiler.
+            These flags are merged with the target-level defines, with the target-level flags taking precedence.
+            """
+        ),
+    )
+
+    cpp_defines = StrListOption(
```

Comment:
`cxx` vs `cpp`. Although you could have a `cpp_defines` as in "C preprocessor" and have an option that is applicable to both C and C++ code.